src/v/cluster/tm_stm.cc

```diff
@@ -131,6 +131,10 @@ std::optional<tm_transaction> tm_stm::do_get_tx(kafka::transactional_id tx_id) {
 
 ss::future<checked<tm_transaction, tm_stm::op_status>>
 tm_stm::get_tx(kafka::transactional_id tx_id) {
+    auto r = co_await sync(_sync_timeout);
```

Comment:
nit: this is the first(*) time we invoke `sync` from tm_stm and it's used for an intermediate txn step; before we used to sync only as a first step of a request handle. Not that it's wrong but it feels a bit risky, if it works without it, probably it's better to drop

Reply:
I added it for tests, but I thought it will be generally useful going forward since get_tx is now futurized and we have a term check below.